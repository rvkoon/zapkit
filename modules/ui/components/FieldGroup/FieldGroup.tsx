import { ReactNode } from "react";
import cx from "classnames";

export interface IFieldGroupProps {
  label?: string;
  htmlFor: string;
  children: ReactNode;
  className?: string;
}

export function FieldGroup({
  label,
  htmlFor,
  children,
  className,
}: IFieldGroupProps): JSX.Element {
  return (
    <div className={cx("flex flex-col gap-1", className)}>
      {label && (
        <label htmlFor={htmlFor} className="pl-1">
          {label}
        </label>
      )}
      {children}
    </div>
  );
}
