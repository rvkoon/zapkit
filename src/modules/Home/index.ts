export * from './components';
export * from './helpers';
export * from './hooks';
export * from './services';
export * from './types';
export * from './constants';