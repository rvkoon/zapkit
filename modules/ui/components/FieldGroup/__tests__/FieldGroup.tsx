import { FieldGroup, IFieldGroupProps } from "../FieldGroup";
import { RenderResult } from "@testing-library/react";
import { render } from "modules/tests";

describe("FieldGroup component", () => {
  function renderComponent(props: IFieldGroupProps): RenderResult {
    return render(<FieldGroup {...props} />);
  }

  it("should render correctly", () => {
    const { container } = renderComponent({
      htmlFor: "test",
      children: <div></div>,
    });
    expect(container).toMatchSnapshot();
  });
});
