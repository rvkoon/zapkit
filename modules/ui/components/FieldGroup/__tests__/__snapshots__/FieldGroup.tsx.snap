// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldGroup component should render correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-1"
  >
    <div />
  </div>
</div>
`;
